marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -59,7 +62,59 @@ private BulkOutputCallerImpl(OutputEndpointImpl endpoint) {
         @Override
         public void awaitCompletion() {
             logger.trace("output endpoint running endpoint={} work={}", getEndpointPath(), getWorkUnit());
+            setPhase(WorkPhase.RUNNING);
```

Comment:
Should we add a test that the current WorkPhase is INITIALIZING and throw an illegal state exception if it isn't?